sup3r/utilities/era_downloader.py

```diff
@@ -622,53 +624,56 @@ def all_months_exist(cls, year, file_pattern):
         """Check if file has been pruned already."""
-        if keep_variables is None:
-            logger.info('Received keep_variables=None. Skipping pruning.')
+        if prune_variables is None:
+            logger.info('Received prune_variables=None. Skipping pruning.')
             return
         else:
-            logger.info(f'Received keep_variables={keep_variables}.')
+            logger.info(f'Received prune_variables={prune_variables}.')
 
         pruned = True
         with Dataset(infile, 'r') as ds:
             variables = [var for var in ds.variables
                          if var not in ('time', 'latitude', 'longitude')]
             for var in variables:
-                if not any(name in var for name in keep_variables):
+                if not any(name in var for name in prune_variables):
                     logger.info(f'Pruning {var} in {infile}.')
                     pruned = False
         return pruned
 
     @classmethod
-    def prune_output(cls, infile, keep_variables=None):
+    def prune_output(cls, infile, prune_variables=None):
         """Prune output file to keep just single level variables"""
-        if keep_variables is None:
-            logger.info('Received keep_variables=None. Skipping pruning.')
+        if prune_variables is None:
+            logger.info('Received prune_variables=None. Skipping pruning.')
             return
         else:
-            logger.info(f'Received keep_variables={keep_variables}.')
+            logger.info(f'Received prune_variables={prune_variables}.')
 
         logger.info(f'Pruning {infile}.')
         tmp_file = cls.get_tmp_file(infile)
         with Dataset(infile, 'r') as old_ds:
+            keep_vars = [var for var in old_ds.variables
+                         if var not in prune_variables and var not
+                         in ('time', 'latitude', 'longitude', 'level')]
             with Dataset(tmp_file, 'w') as new_ds:
                 new_ds = cls.init_dims(old_ds, new_ds,
                                        ('time', 'latitude', 'longitude'))
-                for var in old_ds.variables:
-                    if any(name in var for name in keep_variables):
-                        old_var = old_ds[var]
-                        vals = old_var[:]
-                        _ = new_ds.createVariable(
-                            var, old_var.dtype, dimensions=old_var.dimensions)
-                        new_ds[var][:] = vals
-                        if hasattr(old_var, 'units'):
-                            new_ds[var].units = old_var.units
-                        if hasattr(old_var, 'standard_name'):
-                            standard_name = old_var.standard_name
-                            new_ds[var].standard_name = standard_name
-                        if hasattr(old_var, 'long_name'):
-                            new_ds[var].long_name = old_var.long_name
+                for var in keep_vars:
```

Comment:
I like this semantic naming much better than working with unnamed list comp's. Nice!